src/mmw/js/src/modeling/controllers.js

```diff
@@ -35,22 +36,84 @@ var ModelingController = {
                 name: 'Untitled Project',
                 created_at: Date.now(),
                 area_of_interest: App.map.get('areaOfInterest'),
-                scenarios: new models.ScenariosCollection([
-                    new models.ScenarioModel({
-                        name: 'Current Conditions',
-                        is_current_conditions: true,
-                        active: true
-                    })
-                ])
+                scenarios: new models.ScenariosCollection()
+            });
+
+            // Poll for model results and set the results in the scenario when done.
+            var pollModel = function(scenario) {
+                // If scenario doesn't have its own task model yet, clone and set it from the project.
+                if (!scenario.get('taskModel')) {
+                    scenario.set('taskModel', $.extend(true, {}, project.get('taskModel')));
+                }
+
+                var taskModel = scenario.get('taskModel'),
+                    runoffModel = scenario.get('results').find(function(model) {
+                    return model.get('name') == 'runoff';
+                    });
+
+                var taskHelper = {
+                    postData: {
+                        model_input: JSON.stringify({
+                            modifications: scenario.get('modifications').toJSON(),
+                            area_of_interest: project.get('area_of_interest')
+                        })
+                    },
+
+                    onStart: function() {
+                        runoffModel.set('polling', true);
+                    },
+
+                    setRunoffResult: function() {
+                        var rawResult = taskModel.get('result');
+                        var result = rawResult == "" ? null : JSON.parse(rawResult);
+                        runoffModel.set({
+                            'result': result,
+                            'polling': false
+                        });
+                    },
+
+                    pollSuccess: function() {
+                        this.setRunoffResult();
+                    },
+
+                    pollFailure: function() {
+                        console.log('Failed to get TR55 results.');
+                        this.setRunoffResult();
+                    },
+
+                    startFailure: function() {
+                        console.log('Failed to start TR55 job.');
+                        this.setRunoffResult();
+                    }
+                };
+
+                taskModel.start(taskHelper);
+            };
+
+            // The debounce interval is an estimate of the amount of time
+            // to start the request and
+            var debouncedPollModel = _.debounce(pollModel, 500);
```

Comment:
With the way this is setup, every scenario is sharing the debounced function. So making changes to one scenario and then making changes to another scenario disregards the results for the initial scenario. Each scenario should operate independently.